### PR TITLE
Update Task Route

### PR DESCRIPTION
**Pull request by sselkef**
** I do not recommend reviewing/pulling yet
Not sure if any of this works, just wanted to get it up for review and ask for help on some changes and further details.

### CODE REVIEW

File: server/src/models/tasks.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    .then(([{id}]) => { // not sure if this works, just copied from createTask (thanks Rohit)
+      newTask['id'] = id;
+      return newTask;
+  })
+    .catch((err) => {
+      throw new Error("Failed to update existing Task " + err);
+    })
+}
+
+// ask for more information on structure if body, if it includes special fields
```
Comment (sselkef) [inline server/src/models/tasks.ts:37]: TODO

File: server/src/models/tasks.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            throw new Error('Failed to create new Task ' + err);
+        })
+}
+
+// I have no idea if any of these work
+export async function updateTask(newTask: any) {
+  return await knex("tasks")
+    .where("id", newTask["id"])
+    // chain updates?
+    .update("info_type", newTask["info_type"])
```
Comment (sselkef) [inline server/src/models/tasks.ts:21]: Not sure if newTask["xyz"] is the correct way of accessing the field

File: server/src/models/tasks.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    .update("due_date", newTask.due_date)
+    .update("created_date", newTask.created_date)
+    .returning("id")
+    .then(([{id}]) => { // not sure if this works, just copied from createTask (thanks Rohit)
+      return newTask;
+  })
+    .catch((err) => {
+      throw new Error("Failed to update existing Task " + err);
+    })
+}
```
Comment (benmelz) [inline server/src/models/tasks.ts:34]:
```suggestion
export async function updateTask(attributes: any) {
  const { id, info: info_attributes, ...attributes } = attributes; // extract id and k-specific fields
  const task = await knex('tasks').where({ id }).update(attributes, "*"); // update the task
  task.info = await knex(task.info_type).where({ id: task.info_id }).update(info_attributes, "*"); // update the associated info
  return task;
}
```

This is the most basic example of what you want to implement here. You should assume that the request body will be a task with the addition of an entry for `info`, which will contain the fields for the associated info. Knowing them all isn't necessary here, you can just forward them to the database.

Comment (benmelz) [reply server/src/models/tasks.ts:34]: Also, `await` and `then` are two ways of doing the same thing (asyncronous JS), so you should be using one or the other.

File: server/db/seeds/default.js [hunk 4]
```diff
@@ -133,3 +134,9 @@ exports.seed = async function (knex) {
     "*"
   );
 };
+
+/*
+const test1 = knex("time_off_requests")
+  .where("id", users[0].id)
+  .update("notes", "test changed notes");
+*/
```
Comment (daniel-melanson) [inline server/db/seeds/default.js:142]:
```suggestion

```

File: server/src/models/users.ts [hunk 5]
```diff
@@ -1,5 +1,9 @@
 import knex from "../pool";
 
 export async function getUserFromEmail(email: string) {
-  return knex("users").where({ email }).first();
+  return await knex("users").where({ email }).first();
+}
+
+export async function checkUserID(userId: number) {
+  return await knex("users").select("email").where("id", id).first();
 }
```
Comment (daniel-melanson) [inline server/src/models/users.ts:9]:
```suggestion
export async function doesUserExist(id: number) {
  const result = await knex("users").count("id").where({ id });

  return result.length > 0;
}
```

File: server/src/models/tasks.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        return await knex("training_assignments")
+          .where("id", newTask.info_id)
+          .update("link", newTask.link)
+          .returning("id")
+          .then(([{id}]) => { // not sure if this works, just copied from createTask (thanks Rohit)
+            return newTask;
+          })
+          .catch((err) => {
+            throw new Error("Failed to update existing Task " + err);
+          })
```
Comment (daniel-melanson) [inline server/src/models/tasks.ts:68]:
```suggestion
          .returning("id");
```

File: server/src/models/tasks.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        .update("start_date", newTask.start_date)
+        .update("end_date", newTask.end_date)
+        .update("notes", newTask.notes)
+        .returning("id")
+        .then(([{id}]) => { // not sure if this works, just copied from createTask (thanks Rohit)
+          return newTask;
+        })
+        .catch((err) => {
+          throw new Error("Failed to update existing Task " + err);
+        })
```
Comment (daniel-melanson) [inline server/src/models/tasks.ts:56]:
```suggestion
        .returning("id");
```

**Pull request closed without merging**